curriculum/challenges/english/25-front-end-development/quiz-dom-manipulation-and-click-event-with-javascript/66edd07682767adff3a6231e.md

```diff
@@ -17,439 +17,555 @@ Answer all of the questions below correctly to pass the quiz.
+Text and comment nodes can be accessed using the `children` property
 
 #### --answer--
 
-Placeholder answer
+The root element is always an html tag.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+How can you use JavaScript to find all HTML anchors with missing hyperlinks ?
 
 #### --distractors--
 
-Placeholder distractor 1
+```js
+document.findAll("link![href]");
+```
 
 ---
 
-Placeholder distractor 2
+```js
+document.find("link:not([href])");
+```
 
 ---
 
-Placeholder distractor 3
+```js
+document.querySelector("a![href]");
+```
 
 #### --answer--
 
-Placeholder answer
+```js
+document.querySelectorAll("a:not([href])");
+```
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+In JavaScript, how can you remove the last child node of a paragraph element with the "target" identifier and replace it with a "That's it." text node ?
 
 #### --distractors--
 
-Placeholder distractor 1
+```js
+const target = document.querySelector("p.target");
+target.remove(target.lastElementChild);
+target.appendChild("That's it");
+```
 
 ---
 
-Placeholder distractor 2
+```js
+const target = document.getElementById("target");
+target.lastElementChild.replaceWith("That's it");
+```
 
 ---
 
-Placeholder distractor 3
+```js
+const target = document.querySelector("p#target");
+target.replaceChild("That's it", target.lastChild);
+```
 
 #### --answer--
 
-Placeholder answer
+```js
+const target = document.getElementById("target");
+target.removeChild(target.lastChild);
+target.append("That's it");
+```
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which one of these affirmations is false ?
 
 #### --distractors--
 
-Placeholder distractor 1
+The `innerHTML` element's property returns all the HTML content inside it as text.
 
 ---
 
-Placeholder distractor 2
+The `innerText` element's property only returns the text content that is shown on screen.
 
 ---
 
-Placeholder distractor 3
+The `textContent` node's property returns all the text content inside it.
 
 #### --answer--
 
-Placeholder answer
+The `htmlContent` node's property only returns HTML content that is shown on screen.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following JavaScript lines do not insert a thematic break at the end of the document body ?
 
 #### --distractors--
 
-Placeholder distractor 1
+```js
+document.body.insertBefore(document.createElement("hr"), null);
+```
 
 ---
 
-Placeholder distractor 2
+```js
+document.body.appendChild(document.createElement("hr"));
+```
 
 ---
 
-Placeholder distractor 3
+```js
+document.body.after(document.createElement("hr"));
+```
 
 #### --answer--
 
-Placeholder answer
+```js
+document.body.append("<hr/>");
+```
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following statements is false ?
 
 #### --distractors--
 
-Placeholder distractor 1
+`window` is an alias for `navigator.window`
+```js
+window === navigator.window // returns true
+```
```

Comment:
````suggestion
`window` is an alias for `navigator.window

```js
window === navigator.window // returns true
```
````
code fences always need to be surrounded by empty lines